#! /bin/sh /usr/share/dpatch/dpatch-run
## listen_address_cmdline_synonym.patch.dpatch by  <devin@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add --listen-address as a synonym for --interface, for compatibility
## DP: with the qpsmtpd-prefork commandline interface

@DPATCH@

diff --git a/qpsmtpd-prefork b/qpsmtpd-prefork
index 18a980f..0905224 100755
--- a/qpsmtpd-prefork
+++ b/qpsmtpd-prefork
@@ -92,20 +92,21 @@ my $user;
 sub usage {
     print <<"EOT";
 Usage: qpsmtpd-prefork [ options ]
---quiet             : Be quiet (even errors are suppressed)
---version           : Show version information
---debug             : Enable debug output
---interface addr    : Interface daemon should listen on (default: $d_addr)
---port int          : TCP port daemon should listen on (default: $d_port)
---max-from-ip int   : Limit number of connections from single IP (default: $maxconnip, 0 to disable)
---children int      : Max number of children that can be spawned (default: $max_children)
---idle-children int : Number of idle children to spawn (default: $idle_children, 0 to disable)
---pretty-child      : Change child process name (default: 0)
---user username     : User the daemon should run as
---pid-file path	    : Path to pid file
---renice-parent int : Subtract value from parent process nice level (default: $re_nice)
---detach            : detach from controlling terminal (daemonize)
---help              : This message
+--quiet              : Be quiet (even errors are suppressed)
+--version            : Show version information
+--debug              : Enable debug output
+--listen-address addr: Listen for connections on the address 'addr' (default: $d_addr);
+                       synonymous with --interface
+--port int           : TCP port daemon should listen on (default: $d_port)
+--max-from-ip int    : Limit number of connections from single IP (default: $maxconnip, 0 to disable)
+--children int       : Max number of children that can be spawned (default: $max_children)
+--idle-children int  : Number of idle children to spawn (default: $idle_children, 0 to disable)
+--pretty-child       : Change child process name (default: 0)
+--user username      : User the daemon should run as
+--pid-file path	     : Path to pid file
+--renice-parent int  : Subtract value from parent process nice level (default: $re_nice)
+--detach             : detach from controlling terminal (daemonize)
+--help               : This message
 EOT
     exit 0;
 }
@@ -116,6 +117,7 @@ GetOptions(
     'version'         => sub { print "Qpsmtpd Daemon - version $VERSION\n"; exit 0; },
     'debug'           => \$debug,
     'interface=s'     => \$d_addr,
+    'listen-address=s'=> \$d_addr,
     'port=i'          => \$d_port,
     'max-from-ip=i'   => \$maxconnip,
     'children=i'      => \$max_children,
